packages/profile/src/views/Profile.jsx

```diff
@@ -144,6 +122,66 @@ const Profile = () => {
     })()
   }, [services, user])
 
+  const changeConsentToggle = (enabled) => {
+    setConsentToggleEnabled(enabled)
+
+    try {
+      dispatch({ type: 'consent', payload: enabled })
+    } catch (error) {
+      console.log(error)
+    }
+  }
+
+  const save = async (state, dispatch, e) => {
+    e.preventDefault()
+    setSubmitting(true)
+    setResult(null)
+    setErrorMessage(null)
+
+    const { profiles, members, memberId, profileId } = state
+    const data = Object.keys(state)
+      .filter(key => !['profiles', 'members', 'profileId', 'wallet'].includes(key))
+      .reduce((acc, key) => Object.assign(acc, { [key]: state[key] }), {})
+    console.log(data)
+
+    try {
+      if (profileId && memberId) {
+        const patched = await profiles.patch(profileId, data)
+        console.log('patched', patched)
+        setSubmitting(false)
+        setResult('success')
+        return patched
+      }
+
+      const profile = await profiles.create(data)
+      console.log('new', profile)
+      dispatch({ type: 'profileId', payload: profile.id })
+
+      const member = await members.patch(profile.data.memberId, { profileId: profile.id })
+      console.log(member)
+      setSubmitting(false)
+      setResult('success')
+      // dispatch({ type: 'created', payload: updated })
+    } catch (error) {
+      setSubmitting(false)
+      setResult('error')
+      setErrorMessage(error.message)
+    }
+  }
+
+  let alert
```

Comment:
This should be in its own function component and just return the appropriate alert component.

```
const showAlert = ({submitting, result}) => {
  if (submitting) {
    return <Alert type='...' .../>
  }
  if (result === 'success') {
   ...
  }
```

Having said that, we are now using `submitting, errorMessage, result` to pretty much track the same thing and feels verbose and redunant. 

I think it's worth cleaning this up further since it will be a pattern we expect to use across all apps. 

Personally, I would maybe lean towards using one variable to track the different states: empty, submitting, success / error.

Reply:
That makes sense. I think I still need an extra variable for `errorMessage` though